code/obj/sawflymisc.dm

```diff
@@ -93,39 +110,51 @@
 	inhand_image_icon = 'icons/mob/inhand/tools/omnitool.dmi'
 	icon_state = "sawflycontr"
 
-
 	attack_self(mob/user as mob)
-		for(var/mob/living/critter/robotic/sawfly/S in range(get_turf(src), 5)) // folds active sawflies
+		for (var/mob/living/critter/robotic/sawfly/S in range(get_turf(src), 5)) // folds active sawflies
 			SPAWN(0.1 SECONDS)
 				S.foldself()
 
-		for(var/obj/item/old_grenade/sawfly/S in range(get_turf(src), 4)) // unfolds passive sawflies
-			S.visible_message("<span class='combat'>[S] suddenly springs open as its engine purrs to a start!</span>")
-			S.icon_state = "sawfly1"
-			SPAWN(S.det_time)
-				S.prime()
-		for(var/obj/item/old_grenade/spawner/sawflycluster/S in range(get_turf(src), 4))
-			S.visible_message("<span class='combat'>The [S] suddenly begins beeping as it is primed!</span>")
-			if(S.icon_state=="clusterflyA")
-				S.icon_state = "clusterflyA1"
+		for(var/obj/item/old_grenade/S in range(get_turf(src), 4)) // unfolds passive sawflies
+			if (S.issawfly == TRUE)
+				if ((istype(S, /obj/item/old_grenade/sawflyreused)) || (istype(S, /obj/item/old_grenade/sawfly)))
+					S.visible_message("<span class='combat'>[S] suddenly springs open as its engine purrs to a start!</span>")
+					S.icon_state = "sawfly1"
+					SPAWN(S.det_time)
+						S.prime()
```

Comment:
A lot can happen during a spawn, you should make sure that `S` still exists. and didnt like, fall into a pool of lava